File: ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/transaction/TransactionWithMetadata.java
package net.corda.v5.ledger.common.transaction;

import net.corda.v5.base.annotations.DoNotImplement;
import net.corda.v5.crypto.SecureHash;
import org.jetbrains.annotations.NotNull;

/**
 * Defines metadata properties of transactions common across difference ledger implementations.
 */
@DoNotImplement
public interface TransactionWithMetadata {

    /**
     * Gets the ID of the transaction.
     *
     * @return Returns the ID of the transaction.
     */
    @NotNull
    SecureHash getId();

    /**
     * Gets the metadata for the specified transaction.
     *
     * @return Returns the metadata for the specified transaction.
     */
    @NotNull
    TransactionMetadata getMetadata();
}

File: ledger/ledger-common/src/main/kotlin/net/corda/v5/ledger/common/transaction/TransactionWithMetadata.kt
package net.corda.v5.ledger.common.transaction

import net.corda.v5.base.annotations.DoNotImplement
import net.corda.v5.crypto.SecureHash

/**
 * TransactionWithMetadata contains metadata properties of transactions common across different ledger implementations.
 *
 * * @property signature The signature that was applied.
 */
@DoNotImplement
interface TransactionWithMetadata {
    /**
     * @property id The ID of the transaction.
     */
    val id: SecureHash

    /**
     * @property metadata The metadata for this transaction.
     */
    val metadata: TransactionMetadata
}